File: 프로그래머스/Lv.1/12934. 정수 제곱근 판별/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>
using namespace std;

long long solution(long long n) {
    long long answer = 0;
    long s = sqrt(n);
    
    answer = (s * s == n) ? (s+1) * (s+1) : -1;
    return answer;
}

File: 프로그래머스/Lv.1/12934. 정수 제곱근 판별/정수 제곱근 판별.cs
using System;

public class Solution {
    public long solution(long n) {
        double s = Math.Sqrt(n);
        
        if(s % 1 == 0) {
            return (long) (s+1) * (long) (s+1);
        } else {
            return -1;
        }
    }
}